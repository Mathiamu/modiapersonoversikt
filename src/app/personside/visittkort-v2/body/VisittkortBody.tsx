import * as React from 'react';
import ErrorBoundary from '../../../../components/ErrorBoundary';
import VisuallyHiddenAutoFokusHeader from '../../../../components/VisuallyHiddenAutoFokusHeader';
import { Person } from '../PersondataDomain';
import Familie from './familie/Familie';
import Fullmakter from './fullmakt/Fullmakt';
import Kontaktinformasjon from './kontaktinformasjon/Kontaktinformasjon';
import { Kolonne, VisittkortBodyWrapper } from './VisittkortStyles';
import DeltBosted from './deltBosted/DeltBosted';
import Foreldreansvar from './foreldreansvar/Foreldreansvar';
import TilrettelagtKommunikasjon from './tilrettelagtkommunikasjon/TilrettelagtKommunikasjon';
import Vergemal from './vergemal/Vergemal';
import Sikkerhetstiltak from './sikkerhetstiltak/Sikkerhetstiltak';
import LenkeBrukerprofil from './lenkebrukerprofil/LenkeBrukerprofil';
import { useEffect, useState } from 'react';

interface Props {
    person: Person;
}

function SingleColumnLayout(person: Person) {
    return (
        <Kolonne>
            <Kontaktinformasjon person={person} />
            <Fullmakter fullmakter={person.fullmakt} />
            <Familie person={person} />
            <DeltBosted deltBosted={person.deltBosted} />
            <Foreldreansvar foreldreansvar={person.foreldreansvar} />
            {/*<NavKontorSeksjon />*/}
            <TilrettelagtKommunikasjon tilrettelagtKommunikasjon={person.tilrettelagtKommunikasjon} />
            <Vergemal vergemal={person.vergemal} />
            <Sikkerhetstiltak sikkerhetstiltak={person.sikkerhetstiltak} />
            <LenkeBrukerprofil />
        </Kolonne>
    );
}

function DoubleColumnLayout(person: Person) {
    return (
        <>
            <Kolonne>
                <Kontaktinformasjon person={person} />
                <Fullmakter fullmakter={person.fullmakt} />
                <Familie person={person} />
                <DeltBosted deltBosted={person.deltBosted} />
                <Foreldreansvar foreldreansvar={person.foreldreansvar} />
            </Kolonne>
            <Kolonne>
                {/* {<NavKontorSeksjon />} */}
                <TilrettelagtKommunikasjon tilrettelagtKommunikasjon={person.tilrettelagtKommunikasjon} />
                <Vergemal vergemal={person.vergemal} />
                <Sikkerhetstiltak sikkerhetstiltak={person.sikkerhetstiltak} />
                <LenkeBrukerprofil />
            </Kolonne>
        </>
    );
}

function TripleColumnLayout(person: Person) {
    return (
        <>
            <Kolonne>
                <Kontaktinformasjon person={person} />
                <Fullmakter fullmakter={person.fullmakt} />
            </Kolonne>
            <Kolonne>
                <Familie person={person} />
                <Foreldreansvar foreldreansvar={person.foreldreansvar} />
                <DeltBosted deltBosted={person.deltBosted} />
                <TilrettelagtKommunikasjon tilrettelagtKommunikasjon={person.tilrettelagtKommunikasjon} />
                <Vergemal vergemal={person.vergemal} />
            </Kolonne>
            <Kolonne>
                {/* <NavKontorSeksjon /> */}
                <Sikkerhetstiltak sikkerhetstiltak={person.sikkerhetstiltak} />
                <LenkeBrukerprofil />
            </Kolonne>
        </>
    );
}

function VisittkortBody({ person }: Props) {
    const [width, setWidth] = useState(0);

    useEffect(() => {
        function updateWidth() {
            setWidth(window.innerWidth);
        }
        window.addEventListener('resize', updateWidth);
        updateWidth();
        return () => window.removeEventListener('resize', updateWidth);
    }, []);

    function getColumnLayout(antallKolonner: number) {
        if (antallKolonner <= 1) {
            return SingleColumnLayout(person);
        } else if (antallKolonner === 2) {
            return DoubleColumnLayout(person);
        }
        return TripleColumnLayout(person);
    }

    const maxColumnWidth = 275;
    const numberOfColumns = Math.floor(width / maxColumnWidth);
    const columnLayOut = getColumnLayout(numberOfColumns);

    return (
        <ErrorBoundary>
            <VisittkortBodyWrapper role="region" aria-label="Visittkortdetaljer">
                <VisuallyHiddenAutoFokusHeader tittel="Visittkortdetaljer" />
                {columnLayOut}
            </VisittkortBodyWrapper>
        </ErrorBoundary>
    );
}

export default VisittkortBody;
