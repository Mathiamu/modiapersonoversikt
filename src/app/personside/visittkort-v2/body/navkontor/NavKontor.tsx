import * as React from 'react';
import { Fragment } from 'react';
import styled from 'styled-components/macro';
import { Feilmelding, Normaltekst } from 'nav-frontend-typografi';
import EtikettGraa from '../../../../../components/EtikettGraa';
import VisittkortElement from '../VisittkortElement';
import NavLogo from '../../../../../svg/NavLogo';
import { VisittkortGruppe } from '../VisittkortStyles';
import { Enhet, InformasjonElement, Publikumsmottak as PublikumsmottakInterface } from '../../PersondataDomain';
import AdresseInfo from '../AdresseInfo';
import { capitalizeName } from '../../../../../utils/string-utils';
import { harFeilendeSystemer } from '../../harFeilendeSystemer';
import { mapUgyldigGT } from '../../visittkort-utils';
import baseurls from '../../../../../rest/resources/baseurlsResource';

const ApningstiderListe = styled.dl`
    margin: initial;
    padding: initial;
    list-style: none;
    display: flex;
    flex-flow: row wrap;
    dt {
        flex: 1 1 50%;
    }
    dd {
        flex: 2 2 50%;
        margin: 0;
    }
`;

interface Props {
    feilendeSystemer: Array<InformasjonElement>;
    navEnhet: Enhet | null;
    geografiskTilknytning: string | null;
}

function PublikumsmottakKontaktInfo(props: { publikumsmottak: PublikumsmottakInterface }) {
    const apningstider = props.publikumsmottak.apningstider.map((apningstid) => {
        return (
            <Fragment key={apningstid.ukedag}>
                <dt>
                    <Normaltekst>{capitalizeName(apningstid.ukedag)}</Normaltekst>
                </dt>
                <dd>
                    <Normaltekst>{apningstid.apningstid}</Normaltekst>
                </dd>
            </Fragment>
        );
    });

    return (
        <div>
            <br />
            <EtikettGraa>Besøksadresse</EtikettGraa>
            <AdresseInfo adresse={props.publikumsmottak.besoksadresse} />
            <br />
            <EtikettGraa>Åpningstider</EtikettGraa>
            <ApningstiderListe>{apningstider}</ApningstiderListe>
            <br />
        </div>
    );
}

function Publikumsmottak(props: { publikumsmottak: PublikumsmottakInterface[] }) {
    const publikumsmottak = props.publikumsmottak.firstOrNull();
    if (!publikumsmottak) {
        return <Normaltekst>Ingen publikumsmottak</Normaltekst>;
    }

    const flerePublikumsmottak =
        props.publikumsmottak.length > 1 ? (
            <>
                <Normaltekst>Det finnes flere publikumsmottak</Normaltekst>
                <br />
            </>
        ) : null;

    return (
        <>
            <PublikumsmottakKontaktInfo publikumsmottak={publikumsmottak} />
            {flerePublikumsmottak}
        </>
    );
}

function NavKontor({ feilendeSystemer, navEnhet, geografiskTilknytning }: Props) {
    const baseUrlResource = baseurls.useFetch();
    const baseUrl = baseUrlResource.data?.norg2Frontend ?? '';

    if (harFeilendeSystemer(feilendeSystemer, InformasjonElement.NORG_NAVKONTOR)) {
        return (
            <VisittkortGruppe tittel={'NAV-kontor'}>
                <VisittkortElement beskrivelse={'Ukjent NAV-kontor'} ikon={<NavLogo />}>
                    <Feilmelding>Feilet ved uthenting av informasjon om NAV-kontor</Feilmelding>
                </VisittkortElement>
            </VisittkortGruppe>
        );
    }

    if (geografiskTilknytning === null) {
        return null;
    } else if (navEnhet === null) {
        return (
            <VisittkortGruppe tittel={'NAV-kontor'}>
                <VisittkortElement beskrivelse={mapUgyldigGT(geografiskTilknytning)} ikon={<NavLogo />}>
                    <Feilmelding>Fant ikke geografisk tilknyttning for bruker</Feilmelding>
                </VisittkortElement>
            </VisittkortGruppe>
        );
    }

    const beskrivelse = `${navEnhet?.id} ${navEnhet.navn}`;
    return (
        <VisittkortGruppe tittel={'NAV-kontor'}>
            <VisittkortElement beskrivelse={beskrivelse} ikon={<NavLogo />}>
                <Publikumsmottak publikumsmottak={navEnhet.publikumsmottak} />
                <a
                    href={`${baseUrl}/#/startsok?enhetNr=${navEnhet.id}`}
                    target="_blank"
                    rel="noopener noreferrer"
                    className="lenke"
                >
                    <Normaltekst tag="span">Mer informasjon om kontoret</Normaltekst>
                </a>
            </VisittkortElement>
        </VisittkortGruppe>
    );
}

export default NavKontor;
