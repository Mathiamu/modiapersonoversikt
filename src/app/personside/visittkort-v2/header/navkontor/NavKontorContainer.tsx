import * as React from 'react';
import styled from 'styled-components/macro';
import { Element } from 'nav-frontend-typografi';
import { Bold } from '../../../../../components/common-styled-components';
import { Person } from '../../PersondataDomain';

interface Props {
    person: Person;
}

const NavKontorSection = styled.section`
    margin: 0.5rem 0 0 0;
    display: flex;
    justify-content: flex-end;
    > * {
        white-space: nowrap;
    }
    > *:first-child:after {
        content: '/';
        margin: 0 0.2em;
    }
    > *:last-child {
        display: flex;
        align-items: center;
        margin: 0;
    }
`;

function NavKontorContainer({ person }: Props) {
    if (!person.navEnhet) {
        return (
            <NavKontorSection aria-label="Nav kontor">
                <Element tag="h2">NAV-kontor</Element> <Bold>Ingen enhet</Bold>{' '}
            </NavKontorSection>
        );
    }

    return (
        <NavKontorSection aria-label="Nav kontor">
            <Element tag="h2">NAV-kontor</Element>
            <Element>
                {person.navEnhet.id} {person.navEnhet.navn}
            </Element>
        </NavKontorSection>
    );
}

export default NavKontorContainer;
