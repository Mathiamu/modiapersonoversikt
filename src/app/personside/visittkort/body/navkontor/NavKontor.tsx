import * as React from 'react';
import { Fragment } from 'react';
import styled from 'styled-components';

import Undertekst from 'nav-frontend-typografi/lib/undertekst';

import { NavKontor, PublikumsMottak } from '../../../../../models/navkontor';
import { Klokkeslett } from '../../../../../models/klokkeslett';
import EtikettMini from '../../../../../components/EtikettMini';
import VisittkortElement from '../VisittkortElement';
import NavLogo from '../../../../../svg/NavLogo';
import { BaseUrlsResponse } from '../../../../../models/baseurls';
import { hentBaseUrl } from '../../../../../redux/restReducers/baseurls';
import { ENDASH } from '../../../../../utils/string-utils';

const NameCase = styled.span`
  text-transform: capitalize;
`;

const StyledLenke = styled.a`
  font-size: 0.9em;
`;

const ApningsTiderListe = styled.dl`
  margin: initial;
  padding: initial;
  list-style: none;
  display: flex;
  flex-flow: row wrap;
  dt {
    flex: 1 1 50%;
  }
  dd {
    flex: 2 2 50%;
    margin: 0;
  }
`;

function padMedNull(streng: string) {
    return streng.length === 1 ? `0${streng}` : streng;
}

function klokkeslettToString(klokkeslett: Klokkeslett) {
    return `${padMedNull(klokkeslett.time)}.${padMedNull(klokkeslett.minutt)}`;
}

function publikumsMottakKontaktInfo(publikumsMottak: PublikumsMottak) {
    const apningstider = publikumsMottak.apningstider.map((apningstid) => (
        <Fragment key={apningstid.ukedag}>
            <dt>
                <Undertekst>
                    <NameCase>
                        {apningstid.ukedag.toLowerCase()}
                    </NameCase>
                </Undertekst>
            </dt>
            <dd>
                <Undertekst>
                    {klokkeslettToString(apningstid.apentFra)} {ENDASH} {klokkeslettToString(apningstid.apentTil)}
                </Undertekst>
            </dd>
        </Fragment>
    ));

    const besoksadresse = publikumsMottak.besoksadresse;

    const adresse = besoksadresse
        ? `
            ${besoksadresse.gatenavn} ${besoksadresse.husnummer || '' }${besoksadresse.husbokstav || ''}`
        : 'Adresse ikke funnet';
    const postSted = besoksadresse ? `${besoksadresse.postnummer} ${besoksadresse.poststed}` : '';

    return (
        <div key={adresse}>
            <br/>
            <EtikettMini>Besøksadresse</EtikettMini>
            <Undertekst>{adresse}</Undertekst>
            <Undertekst>{postSted}</Undertekst>
            <br/>
            <EtikettMini>Åpningstider</EtikettMini>
            <ApningsTiderListe>
                {apningstider}
            </ApningsTiderListe>
            <br/>
        </div>
    );
}

function flerePublikumsmottak(antallMottak: number) {
    if (antallMottak > 1) {
        return (
            <>
                <Undertekst>Det finnes flere publikumsmottak</Undertekst>
                <br/>
            </>
        );
    }
    return null;
}

function Publikumsmottak(props: { publikumsmottak: PublikumsMottak[] }) {
    const antallPublikumsmottak = props.publikumsmottak.length;
    if (antallPublikumsmottak === 0) {
        return <Undertekst>Ingen publikumsmottak</Undertekst>;
    }

    const førstePublikumsmottak = props.publikumsmottak[0];
    return (
        <>
            {publikumsMottakKontaktInfo(førstePublikumsmottak)}
            {flerePublikumsmottak(antallPublikumsmottak)}
        </>
    );
}

function navkontorInfo(navKontor: NavKontor, norg2Url: string) {
    return (
        <>
            <Publikumsmottak publikumsmottak={navKontor.publikumsmottak}/>
            <StyledLenke
                href={`${norg2Url}/#/startsok?enhetNr=${navKontor.enhetId}`}
                target="_blank"
                rel="noopener noreferrer"
                className={'lenke'}
            >
                Mer informasjon om kontoret
            </StyledLenke>
        </>
    );
}

function IngenNavKontor() {
    return (
        <VisittkortElement beskrivelse="Ingen enhet" ikon={<NavLogo/>}>
            <br/>
        </VisittkortElement>
    );
}

function NavKontorVisning(props: { navKontor: NavKontor | null, baseUrlsResponse: BaseUrlsResponse }) {
    if (!props.navKontor) {
        return <IngenNavKontor/>;
    }

    const beskrivelse = `${props.navKontor.enhetId} ${props.navKontor.enhetNavn}`;

    const norg2Url = hentNorg2Url(props.baseUrlsResponse);

    return (
        <VisittkortElement beskrivelse={beskrivelse} ikon={<NavLogo/>}>
            {navkontorInfo(props.navKontor, norg2Url)}
        </VisittkortElement>
    );
}

function hentNorg2Url(baseUrlsResponse: BaseUrlsResponse) {
    return hentBaseUrl(baseUrlsResponse, 'norg2-frontend');
}

export default NavKontorVisning;
