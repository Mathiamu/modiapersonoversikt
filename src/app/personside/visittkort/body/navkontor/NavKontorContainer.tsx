import * as React from 'react';
import { connect } from 'react-redux';
import { AppState, Reducer } from '../../../../../redux/reducer';
import Innholdslaster from '../../../../../components/Innholdslaster';
import { NavKontor } from '../../../../../models/navkontor';
import VisittkortElement from '../VisittkortElement';
import NavKontorVisning from './NavKontor';

interface Props {
    navKontorReducer: Reducer<NavKontor>;
}

const navLogo = require('./nav-logo.svg');

class NavKontorContainer extends React.Component<Props> {
    render() {
        return (
            <VisittkortElement beskrivelse="Brukers NavKontor" ikonPath={navLogo}>
                <Innholdslaster avhengigheter={[this.props.navKontorReducer]} spinnerSize={'L'}>
                    <NavKontorVisning navKontor={this.props.navKontorReducer.data}/>
                </Innholdslaster>
            </VisittkortElement>
        );
    }
}

const mapStateToProps = (state: AppState) => {
    return ({
        navKontorReducer: state.brukersNavKontor
    });
};

export default connect(mapStateToProps, null)(NavKontorContainer);
