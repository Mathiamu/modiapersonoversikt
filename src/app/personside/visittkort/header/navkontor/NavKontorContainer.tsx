import * as React from 'react';
import styled from 'styled-components';
import { Normaltekst } from 'nav-frontend-typografi';
import { BrukersNavKontorResponse } from '../../../../../models/navkontor';
import { Bold } from '../../../../../components/common-styled-components';
import RestResourceConsumer from '../../../../../rest/consumer/RestResourceConsumer';

const NavKontorSection = styled.section`
    margin: 0.5rem 0 0 0;
    display: flex;
    justify-content: flex-end;
    > * {
        white-space: nowrap;
    }
    > *:first-child:after {
        content: '/';
        margin: 0 0.2em;
    }
    > *:last-child {
        display: flex;
        align-items: center;
        margin: 0;
    }
`;

const onError = <em>Problemer med Ã¥ hente nav-enhet</em>;

function NavKontorVisning(props: { navKontor: BrukersNavKontorResponse }) {
    if (!props.navKontor) {
        return (
            <Normaltekst>
                <Bold>Ingen enhet</Bold>
            </Normaltekst>
        );
    }

    return (
        <Normaltekst>
            <Bold>
                {props.navKontor.enhetId} {props.navKontor.enhetNavn}
            </Bold>
        </Normaltekst>
    );
}

function NavKontorContainer() {
    return (
        <NavKontorSection aria-label="Nav kontor">
            <Normaltekst tag="h2">
                <Bold>NAV-kontor</Bold>
            </Normaltekst>
            <RestResourceConsumer<BrukersNavKontorResponse>
                getResource={restResources => restResources.brukersNavKontor}
                spinnerSize={'S'}
                returnOnError={onError}
            >
                {navkontor => <NavKontorVisning navKontor={navkontor} />}
            </RestResourceConsumer>
        </NavKontorSection>
    );
}

export default NavKontorContainer;
