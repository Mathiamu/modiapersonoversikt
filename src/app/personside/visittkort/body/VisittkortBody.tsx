import * as React from 'react';
import styled from 'styled-components';
import VisittkortElement from './VisittkortElement';
import { Person } from '../../../../models/person';
import Undertittel from 'nav-frontend-typografi/lib/undertittel';
import Undertekst from 'nav-frontend-typografi/lib/undertekst';
import NavKontorContainer from './navkontor/NavKontorContainer';

const emailPath = require('../../../../resources/svg/email.svg');
const heartPath = require('../../../../resources/svg/heart.svg');
const coinsPath = require('../../../../resources/svg/coins.svg');
const locationPath = require('../../../../resources/svg/location-pin.svg');
const phonePath = require('../../../../resources/svg/phone.svg');
const jentePath = require('../../../../resources/svg/jentebarn.svg');
const guttPath = require('../../../../resources/svg/guttebarn.svg');

interface VisittkortBodyProps {
    person: Person;
}

const VisittkortBodyDiv = styled.div`
  display: flex;
`;

const Kolonne = styled.div`
  flex: 0 0 50%;
  > *:not(:last-child) {
    margin-bottom: 60px;
  }
`;

const GruppeDiv = styled.div`
  > *:not(:last-child):not(:first-child) {
    margin-bottom: 30px;
  }
`;

const PadLeft = styled.span`
  margin-left: 50px;
`;

function InfoGruppe(props: { children: string | JSX.Element | JSX.Element[]; tittel: string; }) {
    return (
        <GruppeDiv>
            <Undertittel>
                <PadLeft>{props.tittel}</PadLeft>
            </Undertittel>
            {props.children}
        </GruppeDiv>
    );
}

function VisittkortBody({ person }: VisittkortBodyProps) {
    return (
        <VisittkortBodyDiv>
            <Kolonne>
                <InfoGruppe tittel={'Kontakt'}>
                    <VisittkortElement beskrivelse="Postadresse Folkeregistrert" ikonPath={locationPath}>
                        <Undertekst>
                            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
                            Facilis neque nobis sint tempora. Quos, tenetur!
                        </Undertekst>
                    </VisittkortElement>
                    <VisittkortElement beskrivelse="Postadresse Midlertidig Norge" ikonPath={locationPath}>
                        <Undertekst>
                            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
                            Et, ipsum.
                        </Undertekst>
                    </VisittkortElement>
                    <VisittkortElement beskrivelse="Epost" ikonPath={emailPath}>
                        <Undertekst>
                            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
                            Exercitationem facere facilis perspiciatis.
                        </Undertekst>
                    </VisittkortElement>
                    <VisittkortElement beskrivelse="Telefon" ikonPath={phonePath}>
                        <Undertekst>
                            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
                            Blanditiis illo in laboriosam magni mollitia quae?
                        </Undertekst>
                    </VisittkortElement>
                    <VisittkortElement beskrivelse="Kontonummer" ikonPath={coinsPath}>
                        <Undertekst>
                            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
                        </Undertekst>
                    </VisittkortElement>
                </InfoGruppe>
            </Kolonne>
            <Kolonne>
                <InfoGruppe tittel={'Familie'}>
                    <VisittkortElement utenTittel={true} beskrivelse="Sivilstatus" ikonPath={heartPath}>
                        <Undertekst>
                            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
                            Adipisci dignissimos eius modi natus praesentium unde velit.
                        </Undertekst>
                    </VisittkortElement>
                    <VisittkortElement utenTittel={true} beskrivelse="Barn" ikonPath={jentePath}>
                        <Undertekst>
                            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
                        </Undertekst>
                    </VisittkortElement>
                    <VisittkortElement utenTittel={true} beskrivelse="Barn" ikonPath={guttPath}>
                        <Undertekst>
                            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
                        </Undertekst>
                    </VisittkortElement>
                </InfoGruppe>
                <InfoGruppe tittel={'Brukers NavKontor'}>
                    <NavKontorContainer />
                </InfoGruppe>
            </Kolonne>
        </VisittkortBodyDiv>
    );
}

export default VisittkortBody;
