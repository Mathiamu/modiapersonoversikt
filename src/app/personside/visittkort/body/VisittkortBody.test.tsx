import * as React from 'react';
import * as renderer from 'react-test-renderer';
import VisittkortBody from './VisittkortBody';
import { testStore } from '../../../../setupTests';
import { Provider } from 'react-redux';
import { getMockNavKontor } from '../../../../mock/navkontor-mock';
import { actionNames } from '../../../../redux/navkontor';
import { aremark } from '../../../../mock/person/aremark';

test('viser info om bruker i visittkortbody', () => {
    const visittkortbody = renderer.create(
        <Provider store={testStore}><VisittkortBody person={aremark} /></Provider>
    );

    testStore.dispatch({type: actionNames.OK, data: getMockNavKontor( '0118')});

    const json = visittkortbody.toJSON();
    expect(json).toMatchSnapshot();
});
