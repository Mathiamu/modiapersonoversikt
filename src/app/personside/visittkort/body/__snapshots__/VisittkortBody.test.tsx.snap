// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viser info om bruker i visittkortbody 1`] = `
<div
  className="sc-cSHVUG dLfBzr"
>
  <div
    className="sc-kAzzGY hIoXoN"
  >
    <div
      className="sc-chPdSV lAsEh"
    >
      <p
        className="typo-element"
      >
        <span
          className="sc-kgoBCf gvexpI"
        >
          Kontakt
        </span>
      </p>
      <div
        className="sc-dnqmqq lhVNBn"
      >
        <div
          className="sc-iwsKbI jYIAlx"
        >
          <img
            alt="Postadresse Folkeregistrert"
            src="location-pin.svg"
            title="Postadresse Folkeregistrert"
          />
        </div>
        <div
          className="sc-gZMcBi ikPRxU"
        >
          <p
            className="typo-etikett-liten"
          >
            <span
              className="sc-gqjmRU dugoZX"
            >
              Postadresse Folkeregistrert
            </span>
          </p>
          <p
            className="typo-undertekst"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Facilis neque nobis sint tempora. Quos, tenetur!
          </p>
        </div>
      </div>
      <div
        className="sc-dnqmqq lhVNBn"
      >
        <div
          className="sc-iwsKbI jYIAlx"
        >
          <img
            alt="Postadresse Midlertidig Norge"
            src="location-pin.svg"
            title="Postadresse Midlertidig Norge"
          />
        </div>
        <div
          className="sc-gZMcBi ikPRxU"
        >
          <p
            className="typo-etikett-liten"
          >
            <span
              className="sc-gqjmRU dugoZX"
            >
              Postadresse Midlertidig Norge
            </span>
          </p>
          <p
            className="typo-undertekst"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Et, ipsum.
          </p>
        </div>
      </div>
      <div
        className="sc-dnqmqq lhVNBn"
      >
        <div
          className="sc-iwsKbI jYIAlx"
        >
          <img
            alt="Epost Kontakt- og reservasjonsregisteret"
            src="email.svg"
            title="Epost Kontakt- og reservasjonsregisteret"
          />
        </div>
        <div
          className="sc-gZMcBi ikPRxU"
        >
          <p
            className="typo-etikett-liten"
          >
            <span
              className="sc-gqjmRU dugoZX"
            >
              Epost Kontakt- og reservasjonsregisteret
            </span>
          </p>
          <svg
            aria-label="Laster innhold"
            className="spinner spinner--l"
            focusable="false"
            height={32}
            kind="spinner"
            onClick={undefined}
            preserveAspectRatio="xMidYMid"
            style={undefined}
            viewBox="0 0 100 100"
            width={32}
          >
            <title>
              Venter...
            </title>
            <circle
              cx="50"
              cy="50"
              fill="none"
              r="40"
              stroke="#bdbab7"
              strokeWidth="10"
              xmlns="http://www.w3.org/2000/svg"
            />
            <circle
              cx="50"
              cy="50"
              fill="none"
              r="40"
              stroke="#7f756c"
              strokeDasharray="50 155"
              strokeLinecap="round"
              strokeWidth="10"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-dnqmqq lhVNBn"
      >
        <div
          className="sc-iwsKbI jYIAlx"
        >
          <img
            alt="Telefon Kontakt- og reservasjonsregisteret"
            src="phone.svg"
            title="Telefon Kontakt- og reservasjonsregisteret"
          />
        </div>
        <div
          className="sc-gZMcBi ikPRxU"
        >
          <p
            className="typo-etikett-liten"
          >
            <span
              className="sc-gqjmRU dugoZX"
            >
              Telefon Kontakt- og reservasjonsregisteret
            </span>
          </p>
          <svg
            aria-label="Laster innhold"
            className="spinner spinner--l"
            focusable="false"
            height={32}
            kind="spinner"
            onClick={undefined}
            preserveAspectRatio="xMidYMid"
            style={undefined}
            viewBox="0 0 100 100"
            width={32}
          >
            <title>
              Venter...
            </title>
            <circle
              cx="50"
              cy="50"
              fill="none"
              r="40"
              stroke="#bdbab7"
              strokeWidth="10"
              xmlns="http://www.w3.org/2000/svg"
            />
            <circle
              cx="50"
              cy="50"
              fill="none"
              r="40"
              stroke="#7f756c"
              strokeDasharray="50 155"
              strokeLinecap="round"
              strokeWidth="10"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-dnqmqq lhVNBn"
      >
        <div
          className="sc-iwsKbI jYIAlx"
        >
          <img
            alt="Kontonummer"
            src="coins.svg"
            title="Kontonummer"
          />
        </div>
        <div
          className="sc-gZMcBi ikPRxU"
        >
          <p
            className="typo-etikett-liten"
          >
            <span
              className="sc-gqjmRU dugoZX"
            >
              Kontonummer
            </span>
          </p>
          <p
            className="typo-undertekst"
          >
            Ingen kontonummer registrert
          </p>
        </div>
      </div>
    </div>
    <div
      className="sc-chPdSV lAsEh"
    >
      <p
        className="typo-element"
      >
        <span
          className="sc-kgoBCf gvexpI"
        >
          Familie
        </span>
      </p>
      <div
        className="sc-dnqmqq lhVNBn"
      >
        <div
          className="sc-iwsKbI jYIAlx"
        >
          <img
            alt="Sivilstand"
            src="heart.svg"
            title="Sivilstand"
          />
        </div>
        <div
          className="sc-gZMcBi ikPRxU"
        >
          <p
            className="typo-etikett-liten"
          >
            <span
              className="sc-gqjmRU dugoZX"
            >
              Sivilstand
            </span>
          </p>
          <p
            className="typo-undertekst-bold"
          >
            Gift
          </p>
        </div>
      </div>
    </div>
    <div
      className="sc-chPdSV lAsEh"
    >
      <p
        className="typo-element"
      >
        <span
          className="sc-kgoBCf gvexpI"
        >
          Navkontor
        </span>
      </p>
      <div
        className="sc-dnqmqq lhVNBn"
      >
        <div
          className="sc-iwsKbI jYIAlx"
        >
          <img
            alt="Brukers NavKontor"
            src="nav-logo.svg"
            title="Brukers NavKontor"
          />
        </div>
        <div
          className="sc-gZMcBi ikPRxU"
        >
          <p
            className="typo-etikett-liten"
          >
            <span
              className="sc-gqjmRU dugoZX"
            >
              Brukers NavKontor
            </span>
          </p>
          <p
            className="typo-undertekst"
          >
            Nav Aremark
             
            0118
          </p>
          <div>
            <p
              className="typo-undertekst"
            >
              
            Rådhuset 6A
            </p>
            <p
              className="typo-undertekst"
            >
              1798 Aremark
            </p>
            <br />
            <p
              className="typo-etikett-liten"
            >
              Åpningstider
            </p>
            <dl
              className="sc-jzJRlG jDxxvr"
            >
              <dt>
                <p
                  className="typo-undertekst"
                >
                  <span
                    className="sc-fjdhpX chqHmI"
                  >
                    mandag
                  </span>
                </p>
              </dt>
              <dd>
                <p
                  className="typo-undertekst"
                >
                  09.00
                   - 
                  14.30
                </p>
              </dd>
              <dt>
                <p
                  className="typo-undertekst"
                >
                  <span
                    className="sc-fjdhpX chqHmI"
                  >
                    tirsdag
                  </span>
                </p>
              </dt>
              <dd>
                <p
                  className="typo-undertekst"
                >
                  09.00
                   - 
                  14.30
                </p>
              </dd>
              <dt>
                <p
                  className="typo-undertekst"
                >
                  <span
                    className="sc-fjdhpX chqHmI"
                  >
                    onsdag
                  </span>
                </p>
              </dt>
              <dd>
                <p
                  className="typo-undertekst"
                >
                  09.00
                   - 
                  14.30
                </p>
              </dd>
              <dt>
                <p
                  className="typo-undertekst"
                >
                  <span
                    className="sc-fjdhpX chqHmI"
                  >
                    torsdag
                  </span>
                </p>
              </dt>
              <dd>
                <p
                  className="typo-undertekst"
                >
                  09.00
                   - 
                  14.30
                </p>
              </dd>
              <dt>
                <p
                  className="typo-undertekst"
                >
                  <span
                    className="sc-fjdhpX chqHmI"
                  >
                    fredag
                  </span>
                </p>
              </dt>
              <dd>
                <p
                  className="typo-undertekst"
                >
                  09.00
                   - 
                  14.30
                </p>
              </dd>
            </dl>
            <br />
          </div>
          <a
            className="sc-jTzLTM gGmsqn"
            href="/norg2-frontend/#/startsok?enhetNr=0118"
            rel="noopener noreferrer"
            target="_blank"
          >
            Mer informasjon om 
            Nav Aremark
          </a>
        </div>
      </div>
    </div>
    <div
      className="sc-dnqmqq lhVNBn"
    >
      <div
        className="sc-iwsKbI jYIAlx"
      >
        <img
          alt="Bruker er under vergemål"
          src="location-pin.svg"
          title="Bruker er under vergemål"
        />
      </div>
      <div
        className="sc-gZMcBi ikPRxU"
      >
        <p
          className="typo-etikett-liten"
        >
          <span
            className="sc-gqjmRU dugoZX"
          >
            Bruker er under vergemål
          </span>
        </p>
        <p
          className="typo-undertekst"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Facilis neque nobis sint tempora. Quos, tenetur!
        </p>
      </div>
    </div>
    <div
      className="sc-dnqmqq lhVNBn"
    >
      <div
        className="sc-iwsKbI jYIAlx"
      >
        <img
          alt="Sikkerhetstiltak"
          src="location-pin.svg"
          title="Sikkerhetstiltak"
        />
      </div>
      <div
        className="sc-gZMcBi ikPRxU"
      >
        <p
          className="typo-etikett-liten"
        >
          <span
            className="sc-gqjmRU dugoZX"
          >
            Sikkerhetstiltak
          </span>
        </p>
        <p
          className="typo-undertekst"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Facilis neque nobis sint tempora. Quos, tenetur!
        </p>
      </div>
    </div>
  </div>
</div>
`;
